### PR TITLE
fix: Improve UI of the login screen

### LINKED ISSUE #1671: Improve UI of the Login screen
**Actual Behaviour**

There are different types of fonts used for same context, spacing issues and so on, that need to be fixed.

**Expected Behaviour**

Improve the UI of the login screen and use the same color scheme as for the entire project.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by aleronarjun**
Fixes #1671 

Changes: Different types of fonts used for the same context have been corrected, and a proper text hierarchy has been adopted. Color scheme has been made in line with rest of the app as well. The sign-up activity has been also slightly modified.

### CODE REVIEW

**Comment by arundhati24** [comment]
@aleronarjun Please read [this](https://chris.beams.io/posts/git-commit/) article and follow commit guidelines.

File: app/src/main/res/layout-land/activity_login.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_gravity="center"
-        android:layout_weight=".5"
+        android:layout_weight="0.5"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: unrelated change.

File: app/src/main/res/layout-land/activity_login.xml [hunk 2]
```diff
@@ -20,13 +20,14 @@
...
+        android:layout_height="wrap_content"
+        android:layout_weight="3">
 
         <LinearLayout
             android:id="@+id/activity_login"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:layout_weight=".5"
+            android:layout_height="wrap_content"
+
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: Remove extra line.

File: app/src/main/res/layout-land/activity_login.xml [hunk 3]
```diff
@@ -38,6 +39,8 @@
                 android:id="@+id/email"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
+                android:layout_marginLeft="@dimen/margin_large"
+                android:layout_marginRight="@dimen/margin_large"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: When you add margin left and right, also add margin start and end.

File: app/src/main/res/layout-land/activity_login.xml [hunk 4]
```diff
@@ -70,6 +75,8 @@
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:layout_marginBottom="5dp"
+                android:layout_marginLeft="@dimen/margin_large"
+                android:layout_marginRight="@dimen/margin_large"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: same.

File: app/src/main/res/layout-land/activity_login.xml [hunk 5]
```diff
@@ -91,48 +98,85 @@
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:layout_gravity="center"
+                android:layout_marginLeft="@dimen/margin_large"
+                android:layout_marginRight="@dimen/margin_large"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: same.

File: app/src/main/res/layout-land/activity_login.xml [hunk 6]
```diff
@@ -91,48 +98,85 @@
...
+                android:orientation="horizontal"
+                android:padding="@dimen/margin_medium">
+
+                <TextView
+                    android:id="@+id/signUp"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_horizontal"
+                    android:layout_marginTop="@dimen/margin_small"
+                    android:layout_weight="1"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: What was the need of adding layout_weight attribute here?

File: app/src/main/res/layout-land/activity_login.xml [hunk 7]
```diff
@@ -91,48 +98,85 @@
...
+                    android:background="?selectableItemBackground"
+                    android:gravity="center"
+                    android:padding="@dimen/padding_medium"
+                    android:text="@string/activity_login_sign_up"
+                    android:textSize="@dimen/text_size_medium"
+                    android:textStyle="bold|italic" />
+
+                <View
+                    android:layout_width="1dp"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: Hardcoded dimen.

File: app/src/main/res/layout-land/activity_login.xml [hunk 8]
```diff
@@ -91,48 +98,85 @@
...
+                    android:layout_marginTop="@dimen/margin_medium"
+                    android:background="@color/blue_grey_300" />
+
+                <TextView
+                    android:id="@+id/forgotPassword"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_horizontal"
+                    android:layout_marginTop="@dimen/margin_small"
+                    android:layout_weight="1"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: What was the need of layout_weight attribute?

**Comment by arundhati24** [comment]
Please check the font that has been used in skills listing fragment and use similar fonts and text sizes here as well.

**Comment by arundhati24** [comment]
I think you can move forgot password below the password field and add another button for sign up. Skip login can be kept at the bottom.
@iamareebjamal  @akshatnitd What do you suggest?

**Comment by aleronarjun** [comment]
@arundhati24 thanks for reviewing and suggesting changes, will do them at the earliest.
Also, I tried doing exactly what you suggested with the forgot password/sign up, and IMHO, this layout looked the cleanest, however I will make that change too, if you insist.

**Comment by aleronarjun** [comment]
> Please check the font that has been used in skills listing fragment and use similar fonts and text sizes here as well.

Will change all text sizes to "moderate".
On the font style side, I see sans-serif everywhere, or do you mean the styling? (style, colour, etc)

**Comment by akshatnitd** [comment]
@arundhati24 @aleronarjun Here are few changes that I suggest - 
* The background color of the page should be white, as the entire app UI after login has been shifted to white.
* The content on the login screen should be placed vertically on the center on the page.
* The option for Custom server can be removed from here, as it seems a bit unrelated here, it should be implemented on the Settings Activity, as done for other SUSI clients.
* Forgot password should be kept just below the Password field.
* `or` and `|`  can then be removed.

@iamareebjamal Should we consider sticking to only the Portrait layout for the Login/Signup screens?

File: app/src/main/res/layout/activity_login.xml [hunk 9]
```diff
@@ -82,47 +87,84 @@
...
+                android:layout_weight="1"
+                android:background="?selectableItemBackground"
+                android:gravity="center"
+                android:padding="@dimen/padding_medium"
+                android:text="@string/activity_login_sign_up"
+                android:textSize="@dimen/text_size_medium"
+                android:textStyle="bold|italic" />
+
+            <View
+                android:layout_width="1dp"
```
Comment (batbrain7) [inline app/src/main/res/layout/activity_login.xml]: Hardcoded Dimension

Comment (aleronarjun) [reply app/src/main/res/layout/activity_login.xml]: Noted.

**Comment by aleronarjun** [comment]
Did all the other changes except:
- Removing the custom server option. Maybe another issue can be raised first to implement it in the Settings Activity before removing it from here?
- Removing the landscape layouts. Need a go-ahead for this @akshatnitd 

New activities:

screenshot_1536951292
screenshot_1536951304
screenshot_1536951316

**Comment by aleronarjun** [comment]
Did everything @arundhati24 
Sorry for the hold up, got mid-sems going on.

**Comment by iamareebjamal** [comment]
Please follow the commit guidelines

File: app/src/main/res/layout-land/activity_login.xml [hunk 10]
```diff
@@ -53,23 +58,46 @@
...
                     android:hint="@string/password"
                     android:imeOptions="actionGo"
                     android:inputType="textPassword" />
+
+                <TextView
+                    android:id="@+id/forgotPassword"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="left"
+                    android:layout_weight="1"
```
Comment (arundhati24) [inline app/src/main/res/layout-land/activity_login.xml]: What is the need of defining layout_weight here?

**Comment by aleronarjun** [comment]
Here is the current UI for login and signup activities (Similar changes for landscape):


screenshot_1537802672

screenshot_1537802678

**Comment by arundhati24** [comment]
@aleronarjun Please change the color of the line in the edit text box also to white.

**Comment by aleronarjun** [comment]
@arundhati24 @mariobehling 

Current UI (Also added logic for enable/disable of the logIn button) :

screenshot_1538665375

screenshot_1538665379

**Comment by aleronarjun** [comment]
Please review this PR, @mariobehling @arundhati24 @iamareebjamal . Thanks.

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 11]
```diff
@@ -121,6 +123,54 @@ class LoginActivity : AppCompatActivity(), ILoginView {
...
+            override fun afterTextChanged(s: Editable?) {
+
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                if(s.toString().trim()!=""){
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Add spaces around parentheses. Please reformat the code for all the changes that you have made.

Comment (aleronarjun) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Sure.

**Comment by arundhati24** [comment]
@aleronarjun Please check all reviews, squash your commits and also please go through this [article](https://chris.beams.io/posts/git-commit/) and follow commit guidelines.

**Comment by aleronarjun** [comment]
@arundhati24 could you help me resolve the conflicting file please? Thanks.

**Comment by aleronarjun** [comment]
Please review @mariobehling . Thank you.

File: app/src/debug/AndroidManifest.xml [hunk 12]
```diff
@@ -43,12 +43,12 @@
         <activity
             android:name=".signup.SignUpActivity"
             android:theme="@style/SignUp_Theme"
-            android:windowSoftInputMode="stateVisible|adjustResize" />
```
Comment (iamareebjamal) [inline app/src/debug/AndroidManifest.xml]: Why is this changed?

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 13]
```diff
@@ -121,6 +123,54 @@ class LoginActivity : AppCompatActivity(), ILoginView {
...
+        logInbuttonEnable()
+    }
+
+    private fun logInbuttonEnable() {
+        var emailEmpty = true
+        var passwordEmpty = true
+        logIn.isEnabled = false
+        emailInput.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Add comments in empty blocks

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 14]
```diff
@@ -121,6 +123,54 @@ class LoginActivity : AppCompatActivity(), ILoginView {
...
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                if (s.toString().trim() != "") {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Rename s to a meaningful word

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 15]
```diff
@@ -121,6 +123,54 @@ class LoginActivity : AppCompatActivity(), ILoginView {
...
+                    emailEmpty = false
+                    logIn.isEnabled = !(emailEmpty || passwordEmpty)
+                } else {
+                    emailEmpty = true
+                    logIn.isEnabled = !(emailEmpty || passwordEmpty)
+                }
+            }
+
+        })
+        passwordInput.addTextChangedListener(object : TextWatcher {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Add newline above

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 16]
```diff
@@ -121,8 +121,8 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
     }
 
     private fun setReportButton() {
-
-        if (PrefManager.getToken() != null) {
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Remove

File: app/src/debug/AndroidManifest.xml (hunk 12 above)
Comment (aleronarjun) [reply app/src/debug/AndroidManifest.xml]: Before changing this, the keyboard would automatically popup (with focus on the first EditText), without the user doing anything, and move all UI elements upwards.
